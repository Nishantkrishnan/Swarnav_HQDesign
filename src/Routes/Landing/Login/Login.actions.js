import {
  ACTION_LOGIN_REQUEST,
  ACTION_STORE_USER_PROFILE_INFO,
  ACTION_LOGOUT,
  RESET_STORE,
} from './LoginPageConstants';

export function logIn(mobileNo, password) {
  return {
    type: ACTION_LOGIN_REQUEST,
    payload: { mobile_no: mobileNo, password },
  };
}

export function storeUserProfileInfo(response) {
  return {
    type: ACTION_STORE_USER_PROFILE_INFO,
    payload: { response },
  };
}

export function logOut({ userToken }) {
  return {
    type: ACTION_LOGOUT,
    payload: userToken,
  };
}

export function clearStore() {
  return {
    type: RESET_STORE,
  };
}
