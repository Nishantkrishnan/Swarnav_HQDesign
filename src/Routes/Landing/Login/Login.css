.wrapper {
  height: 100vh;
  position: relative;
  /* background: url('../../images/shopkickLogin.jpg') no-repeat; */
  background-size: cover;
}

.container {
  width: 420px;
  height: 468px;
  padding: 30px;
  background: #ffffff;
  border-radius: 5px;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 100px;
  margin: auto;
}

.logo {
  display: block;
  /* background: url('../../../images/logo.svg') no-repeat center; */
  background-size: 291px 53px;
  width: 291px;
  height: 53px;
  margin: auto;
  text-align: center;
  cursor: default;
}

.fieldsWrapper {
  margin-bottom: 55px;
}

.inputField {
  width: 100%;
  height: 50px;
  background: #efefef;
  display: block;
  margin-top: 40px;
  border-radius: 5px;
  color: #000000;
  padding: 15px;
  font-size: 13px;
}

.inputField:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 1000px #efefef inset;
}

.link {
  margin: 3px 0 0;
  float: right;
  text-decoration: none !important;
  color: #58bce4 !important;
  font-size: 12px !important;
}

.loginButton {
  width: 100%;
  height: 50px;
  background: #0098d6;
  color: #ffffff;
  font-size: 12px;
  font-weight: 100;
  letter-spacing: 0.3px;
  transition: 100ms all ease-in-out;
}

.remember {
  color: #484747;
  margin: 40px 0 10px;
  text-align: center;
  font-size: 13px;
  cursor: pointer;
}

.loginMessage {
  position: absolute;
  top: 150px;
}

.formSignin {
  width: 100%;
  max-width: 330px;
  padding: 15px;
  margin: 0 auto;
}

.loginBox {
  padding-top: 40px;
  background-color: #f5f5f5;
  text-align: center;
  position: fixed;
  padding: 100px 0 20px;
  margin: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.formSignin .checkbox {
  font-weight: 400;
}

.formSignin .formControl {
  position: relative;
  box-sizing: border-box;
  height: auto;
  padding: 10px;
  font-size: 16px;
  margin-bottom: 5px;
  width: 300px;
}

.formSignin .formControl:focus {
  z-index: 2;
}

.formSignin input[type="phone"] {
  width: 100%;
  margin-bottom: -1px;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.formSignin input[type="password"] {
  width: 100%;
  margin-bottom: 10px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

@media (max-width: 768px) {
  .loginBox {
    padding: 60px 0 0;
  }
}

.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 60px;
  background-color: #f5f5f5;
}
