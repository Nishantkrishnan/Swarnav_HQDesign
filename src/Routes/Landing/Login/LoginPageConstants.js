export const ACTION_LOGIN_REQUEST = '/api/v1/sessions';
export const ACTION_STORE_USER_PROFILE_INFO = '/api/v1/profiles';
export const ACTION_LOGOUT = '/common/logout';
export const RESET_STORE = 'common/resetStore';
