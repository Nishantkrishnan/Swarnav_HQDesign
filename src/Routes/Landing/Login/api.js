import axios from 'axios';
import { authInfo } from '../../../utils/config';

export const loginUser = payload =>
  axios.post(`${authInfo.mainUrl}/api/v1/sessions`, payload)
    .then(res => res.data);

