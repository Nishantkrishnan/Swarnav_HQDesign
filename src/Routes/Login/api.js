import axios from 'axios';
import { authInfo } from '../../utils/config';

// axios.interceptors.request.use(
//   config => {
//     config.headers.Authorization = localStorage.getItem('shopkick-accessToken-remember');
//     return config;
//   },
//   error => Promise.reject(error),
// );


export const loginUser = payload =>
  axios.post(`${authInfo.mainUrl}/api/v1/sessions`, payload)
    .then(res => res.data);

// export const resetPwd = email =>
//   axios.post('/v1/users/forgotpwd', { email })
//     .then(res => res.data);

// export const logout = user_id =>
//   axios.post('/v1/users/logout', { user_id })
//     .then(res => res.data);

// export const getUserDetails = userId =>
//   axios.get(`/v1/users/${userId}`)
//     .then(res => res.data);
