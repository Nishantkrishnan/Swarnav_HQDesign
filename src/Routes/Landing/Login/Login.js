import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Form,
  FormGroup,
  ControlLabel,
  FormControl,
  Checkbox,
  Button,
} from 'react-bootstrap';
import styles from './Login.css';
import { logIn } from './Login.actions';

class Login extends Component {
  static propTypes = {
    dispatch: PropTypes.func,
  };
  static defaultProps = {
    dispatch: f => f,
  };

  constructor(props, context) {
    super(props, context);
    this.login = this.login.bind(this);
  }

  getValidationState = () => { }

  login = () => {
    this.props.dispatch(logIn(this.mobile_no.value, this.password.value));
  }

  render() {
    return (
      <div>
        <Form
          className={styles.formSignin}
          onSubmit={
            (e) => {
              e.preventDefault();
              e.stopPropagation();
              this.login();
            }
          }
        >
          <img className="mb-4" src="../../src/images/hq_final_white-01.svg" alt="" width="150" height="150" />
          <h1 className="h3 mb-3 font-weight-normal">Sign in</h1>

          <FormGroup
            validationState={this.getValidationState()}
          >
            <ControlLabel className="sr-only">Phone</ControlLabel>
            <FormControl
              type="text"
              className={styles.formControl}
              placeholder="Phone / Email"
              inputRef={(ref) => { this.mobile_no = ref; }}
              required=""
              autoFocus="true"
            />

            <ControlLabel className="sr-only">Password</ControlLabel>
            <FormControl
              type="password"
              className={styles.formControl}
              placeholder="Password"
              inputRef={(ref) => { this.password = ref; }}
              required=""
              autoFocus=""
            />

            <Checkbox readOnly>Remember me</Checkbox>
            <Button type="submit" className="btn btn-primary">Sign in</Button>
          </FormGroup>
        </Form>
      </div>
    );
  }
}

export default connect()(Login);
