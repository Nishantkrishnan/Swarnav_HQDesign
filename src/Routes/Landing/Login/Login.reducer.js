import { fromJS } from 'immutable';
import { createReducerFromObject } from '../../../utils/reducerUtils';
import { ACTION_STORE_USER_PROFILE_INFO, RESET_STORE } from './LoginPageConstants';

export const initialState = fromJS({
  error: null,
  loggedIn: false,
  // data: {
  //   userId: localStorage.getItem('userId') || null,
  //   userSessionToken: null,
  // },
  userInfo: null,
});

// export const userDataOnLogin = (state, payload) =>
//   fromJS({
//     error: '',
//     loggedIn: true,
//     data: {
//       userId: payload.response.id,
//       userSessionToken: payload.response.token,
//     },
//     userInfo: null,
//   });

const reducerFunctions = {
  [ACTION_STORE_USER_PROFILE_INFO]: (state, payload) => state.merge({ 
    userInfo: fromJS(payload.response.data),
    loggedIn: true,
  }),
  [RESET_STORE]: state => state.merge({
    loggedIn: false,
    userInfo: null,
  }),
};

const loginReducer = createReducerFromObject(reducerFunctions, initialState);
export default loginReducer;
