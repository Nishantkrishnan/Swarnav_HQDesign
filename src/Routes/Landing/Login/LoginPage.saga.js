import { replace } from 'react-router-redux';
import { fork, put, takeLatest } from 'redux-saga/effects';
import { showMessageAlert } from '../../../containers/MessageAlert/MessageAlert.actions';
import ShowAlert from '../../../containers/ShowAlert/ShowAlert';
import { store } from '../../../index';
import * as actions from './Login.actions';
import * as constants from './LoginPageConstants';
import * as api from './api';

function* loginUser({ payload }) {
  try {
    const response = yield api.loginUser(payload);
    localStorage.setItem('coworks-accessToken-remember', response.data.token);
    yield store.dispatch(actions.storeUserProfileInfo(response));
  } catch (e) {
    ShowAlert(e);
  }
}

function* logout({ payload }) {
  try {
    localStorage.clear();
    yield store.dispatch(actions.clearStore());
    if (payload === null) {
      yield store.dispatch(showMessageAlert({
        message: 'Logged out succesfully!',
        visible: true,
        color: '#61d079',
      }));
    } else {
      yield store.dispatch(showMessageAlert({
        message: 'Your session has expired, please login with your username and password to continue',
        visible: true,
        color: '#e53e3d',
      }));
    }
    yield put(replace('/'));
  } catch (e) {
    ShowAlert(e);
  }
}

export default function* LoginSagas() {
  yield [
    fork(takeLatest, constants.ACTION_LOGIN_REQUEST, loginUser),
    fork(takeLatest, constants.ACTION_LOGOUT, logout),
  ];
}
