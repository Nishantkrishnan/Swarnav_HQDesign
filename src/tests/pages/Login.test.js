/**
 * Created by sireesha on 15/11/17.
 */
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../Routes/Login/Login';

describe('Login Page tests', () => {
  let LoginPageTree;
  let wrapper;
  beforeEach(() => {

  });

  beforeAll(() => {
    LoginPageTree = renderer.create(<Login />);
    wrapper = mount(<Login />);
  });

  test('Snapshot match testing', () => {
    expect(LoginPageTree).toMatchSnapshot();
  });
  test('Component structure testing', () => {
    expect(wrapper.find('Input').node).toBeDefined();
  });
  test('structure testing: Check logo', () => {
    expect(wrapper.find('.logo').node).toBeDefined();
  });
  test('structure testing: Check forgot password link', () => {
    expect(wrapper.find('.link').node).toBeDefined();
  });
  test('structure testing: Check Login Button', () => {
    expect(wrapper.find('Button').node).toBeDefined();
  });
  test('structure testing: Check Checkbox Button', () => {
    expect(wrapper.find('Checkbox').node).toBeDefined();
  });
});
