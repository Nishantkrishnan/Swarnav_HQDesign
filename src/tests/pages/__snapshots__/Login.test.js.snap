// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page tests Snapshot match testing 1`] = `undefined`;
